src/test/java/org/springframework/metrics/instrument/scheduling/ExecutorServiceMetricsTest.java

```diff
@@ -0,0 +1,59 @@
+package org.springframework.metrics.instrument.scheduling;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.springframework.metrics.instrument.Gauge;
+import org.springframework.metrics.instrument.Meter;
+import org.springframework.metrics.instrument.MeterRegistriesProvider;
+import org.springframework.metrics.instrument.MeterRegistry;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+class ExecutorServiceMetricsTest {
+
+  @DisplayName("tp task executor can be instrumented after being initialized")
+  @ParameterizedTest
+  @ArgumentsSource(MeterRegistriesProvider.class)
+  void taskExecutor(MeterRegistry registry) {
```

Comment:
Just validating the scheduler not supporting the decorator. I wouldn't mind deleting it.

 Also, would it make sense to add the task decorator upstream?